### PR TITLE
Setup seed to isolate the different configs

### PR DESCRIPTION
**Pull request by radtriste**
Seed library, main config and branch config becomes independent and can be configured separately in different locations

**Changes**
- [x] main and branch seed jobs should be totally independent via config files  
- [x] common jobs generated via seed library should reference the SEED_AUTHOR and SEED_BRANCH
- [x] prepare release branch should be configurable
     * [x] cut-off kogito-pipelines with optaplanner- or kogito- prefix
         * becomes the base for seed generation
     * [x] setup the branch config file with repo/branches
- [x] Update documentation

**Tests**
- [x] Seed generation
    * [x] Test branch
    * [x] "old branch"
- [x] Release prepare
- [x] Verify trigger jobs config
- [x] Verify common jobs git information => using SEED git info ?
    * [x] PR jobs
    * [x] Update version tools jobs

### CODE REVIEW

File: dsl/seed/jenkinsfiles/Jenkinsfile.release.prepare [hunk 1]
```diff
@@ -0,0 +1,341 @@
+import org.jenkinsci.plugins.workflow.libs.Library
+
+@Library('jenkins-pipeline-shared-libraries')_
+
+import java.util.regex.Pattern
```
Comment (Ginxo) [inline dsl/seed/jenkinsfiles/Jenkinsfile.release.prepare]: why don't you group import statements together?

File: dsl/seed/jenkinsfiles/Jenkinsfile.seed.main [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+        stage('Launch branch seed jobs') {
+            steps {
+                script {
+                    Map parallelJobs = [:]
+                    readSeedConfig().git.branches.collect { it.name }.each { branch ->
+                        parallelJobs.put(branch, { launchBranchSeedJob(branch) })
+                }
+
+                    parallel parallelJobs
```
Comment (Ginxo) [inline dsl/seed/jenkinsfiles/Jenkinsfile.seed.main:154]:
```suggestion
                parallel parallelJobs
```

File: dsl/seed/jenkinsfiles/Jenkinsfile.seed.main [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+            }
+        }
+    }
+}
```
Comment (Ginxo) [inline dsl/seed/jenkinsfiles/Jenkinsfile.seed.main]: is it me or we have an additional curly bracket? :thinking:

**Review by Ginxo — APPROVED** [review]
LGTM, just validating technical side of the implementation. A few minor comments.

File: dsl/seed/jenkinsfiles/Jenkinsfile.release.prepare (hunk 1 above)
Comment (radtriste) [reply dsl/seed/jenkinsfiles/Jenkinsfile.release.prepare]: true. gonna update that.

File: dsl/seed/jenkinsfiles/Jenkinsfile.seed.main (hunk 3 above)
Comment (radtriste) [reply dsl/seed/jenkinsfiles/Jenkinsfile.seed.main]: my formatter messed up. no extra curly bracket, just it did not put the correct indentation. gonna correct that

File: dsl/seed/jenkinsfiles/Jenkinsfile.seed.main (hunk 2 above)
Comment (radtriste) [reply dsl/seed/jenkinsfiles/Jenkinsfile.seed.main:154]: same indentation problem here

**Pull request merged**